lib/connection_string.js

```diff
@@ -500,6 +493,46 @@ function checkTLSOptions(queryString) {
   }
 }
 
+/**
+ * Checks options object if both options are present (any value) will throw an error.
+ *
+ * @param {object} options The options used for options parsing
+ * @param {string} optionKeyA A options key
+ * @param {string} optionKeyB B options key
+ * @throws {MongoParseError}
+ */
+function assertRepelOptions(options, optionKeyA, optionKeyB) {
+  if (options.hasOwnProperty(optionKeyA) && options.hasOwnProperty(optionKeyB)) {
+    throw new MongoParseError(`The \`${optionKeyA}\` option cannot be used with \`${optionKeyB}\``);
+  }
+}
+
+/**
+ * Checks if TLS options are valid
+ *
+ * @param {object} options The options used for options parsing
+ * @throws {MongoParseError}
+ */
+function checkTLSOptions(options) {
+  if (!options) return null;
+
```

Comment:
A small suggestion for improved readability, adding this:

`const check = (a, b) => assertRepelOptions(options, a, b);`

and then replacing `assertRepelOptions(options, ...)` with `check(...)` below allows each check to fit on a single line and is a bit more readable IMO.

Reply:
WOW I like this 👍